packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js

```diff
@@ -172,6 +173,25 @@ const CheckoutPage = props => {
                 </h3>
             );
 
+        const formErrors = [];
+        const paymentMethods = Object.keys(payments);
+
+        // If we don't have an implementation for a method type, ignore it.
+        const isPaymentAvailable = !!availablePaymentMethods.find(({ code }) =>
+            paymentMethods.includes(code)
+        );
```

Comment:
If zero total checkout is enabled, and a user or coupon causes the total to be $0, then `code` will be `free`. As we don't have a payment implementation view for this, we just need to allow the code `free` to mean that payment is available. Making the below changes allows checkout to proceed when a user has entered a coupon/gc that causes the total to be zero.

```suggestion
        // If we have an implementation, or if this is a "zero" checkout,
        // we can allow checkout to proceed.
        const isPaymentAvailable = !!availablePaymentMethods.find(
            ({ code }) => code === 'free' || paymentMethods.includes(code)
        );
```

Reply:
However, in addition to this change, we may want to change the copy here. `free` will only be the code when a GC/coupon is entered causing the total to be zero. 

That means if a user gets to checkout page _before_ adding a gift card or coupon they will first see the error messaging. So we might want to update the message to explain this somehow. cc @schensley